manager/manager.go

```diff
@@ -308,6 +311,7 @@ func (m *Manager) Run(parent context.Context) error {
 	api.RegisterRaftMembershipServer(m.server, proxyRaftMembershipAPI)
 	api.RegisterControlServer(m.localserver, localProxyControlAPI)
 	api.RegisterControlServer(m.server, authenticatedControlAPI)
+	api.RegisterResourceAllocatorServer(m.server, authenticatedResourceAPI)
```

Comment:
I think you do need the proxy. The agent doesn't know which manager is the leader. When it connects to the dispatcher, it connects to any manager, and relies on its raft proxy to redirect the RPCs. I think the same would apply for `ResourceAllocator`.

The node information is forwarded along with forwarded RPCs. `RemoteNode` should work transparently.